### PR TITLE
upup/pkg/fi/ upup/pkg/kutil : simplify code and remove code

### PR DESCRIPTION
**Pull request by tanjunchen**
ref:https://github.com/kubernetes/kops/issues/7800

### CODE REVIEW

File: upup/pkg/fi/cloudup/awsup/aws_cloud.go [hunk 1]
```diff
@@ -908,27 +911,21 @@ func getELBTags(c AWSCloud, loadBalancerName string) (map[string]string, error)
 	request := &elb.DescribeTagsInput{
 		LoadBalancerNames: []*string{&loadBalancerName},
 	}
+	response, err := c.ELB().DescribeTags(request)
+	if err != nil {
+		return nil, fmt.Errorf("error listing tags on %v: %v", loadBalancerName, err)
+	}
 
-	attempt := 0
```
Comment (tanjunchen) [inline upup/pkg/fi/cloudup/awsup/aws_cloud.go]: I am confused that this function of for{} , 
this seems run one,so i think it is ok to remove this code .

File: upup/pkg/fi/cloudup/gcetasks/storagebucketiam.go [hunk 2]
```diff
@@ -147,10 +147,10 @@ func (_ *StorageBucketIam) RenderGCE(t *gce.GCEAPITarget, a, e, changes *Storage
 	return nil
 }
 
-type terraformStorageBucketIam struct {
```
Comment (tanjunchen) [inline upup/pkg/fi/cloudup/gcetasks/storagebucketiam.go]: line 158 refers to this struct , but commented. So I comment this struct not remove it.

**Comment by tanjunchen** [comment]
@justinsb @rifelpet

File: upup/pkg/fi/cloudup/openstacktasks/lblistener.go [hunk 3]
```diff
@@ -69,7 +69,6 @@ func NewLBListenerTaskFromCloud(cloud openstack.OpenstackCloud, lifecycle *fi.Li
 		}
 		listenerTask.Pool = poolTask
 		// TODO: Support Multiple?
-		break
```
Comment (rifelpet) [inline upup/pkg/fi/cloudup/openstacktasks/lblistener.go]: Wouldn't removing this change the `listenerTask.Pool` from being the first item in `lb.Pools` to the last item?

Comment (tanjunchen) [reply upup/pkg/fi/cloudup/openstacktasks/lblistener.go]: Sorry, forget it.
NewLBListenerTaskFromCloud has not comment , so i has some questions about this function. 
This seems  that we need not to  use  for{}  . any advice?

File: upup/pkg/fi/cloudup/alitasks/securitygrouprule.go [hunk 4]
```diff
@@ -49,7 +49,7 @@ func (s *SecurityGroupRule) CompareWithID() *string {
 }
 
 func equalsIgnoreCase(a, b string) bool {
-	if strings.ToLower(a) == strings.ToLower(b) {
+	if strings.EqualFold(a, b) {
```
Comment (hakman) [inline upup/pkg/fi/cloudup/alitasks/securitygrouprule.go]: Seems to me that `equalsIgnoreCase`  just reimplements `strings.EqualFold` and is used only once.
Maybe would be best to just remove it and replace where it's used..

Comment (tanjunchen) [reply upup/pkg/fi/cloudup/alitasks/securitygrouprule.go]: Sure.  I can do it

File: upup/pkg/fi/cloudup/openstacktasks/lblistener.go (hunk 3 above)
Comment (tanjunchen) [reply upup/pkg/fi/cloudup/openstacktasks/lblistener.go]: @hakman any advice?Thanks

Comment (hakman) [reply upup/pkg/fi/cloudup/openstacktasks/lblistener.go]: I would try this:
```diff
--- upup/pkg/fi/cloudup/openstacktasks/lblistener.go.old
+++ upup/pkg/fi/cloudup/openstacktasks/lblistener.go
@@ -66,10 +66,11 @@
 		poolTask, err := NewLBPoolTaskFromCloud(cloud, lifecycle, &pool, find.Pool)
 		if err != nil {
 			return nil, fmt.Errorf("NewLBListenerTaskFromCloud: Failed to create new LBListener task for pool %s: %v", pool.Name, err)
-		}
-		listenerTask.Pool = poolTask
-		// TODO: Support Multiple?
-		break
+		} else {
+			listenerTask.Pool = poolTask
+			// TODO: Support Multiple?
+			break
+		}
 	}
 	if find != nil {
 		// Update all search terms
```

**Comment by rifelpet** [comment]
This looks great, the for loops that make AWS calls wouldnt have been necessary even if they were retrying properly because the AWS client has its own retry + backoff built in, so this definitely simplifies those functions. Thanks!

/lgtm
/approve

**Pull request merged**